File: 3471_find_the_largest_almost_missing_integer/solution.c
// LeetCode 3471 - Find the Largest Almost Missing Integer
// https://leetcode.com/problems/find-the-largest-almost-missing-integer/

int largestInteger(int* nums, int numsSize, int k) {
    int cnt[51];
    for (int i = 0; i < 51; i++) cnt[i] = 0;
    for (int i = 0; i + k <= numsSize; i++) {
        int seen[51] = {0};
        for (int j = i; j < i + k; j++) {
            int x = nums[j];
            if (x >= 0 && x <= 50) seen[x] = 1;
        }
        for (int x = 0; x <= 50; x++) {
            if (seen[x]) cnt[x]++;
        }
    }
    int ans = -1;
    for (int x = 0; x <= 50; x++) {
        if (cnt[x] == 1 && x > ans) ans = x;
    }
    return ans;
}

File: 3471_find_the_largest_almost_missing_integer/solution.rb
# LeetCode 3471 - Find the Largest Almost Missing Integer
# https://leetcode.com/problems/find-the-largest-almost-missing-integer/

# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def largest_integer(nums, k)
  n = nums.length
  cnt = Hash.new(0)
  (0..(n - k)).each do |i|
    seen = {}
    (i...(i + k)).each { |j| seen[nums[j]] = true }
    seen.each_key { |x| cnt[x] += 1 }
  end
  ans = -1
  cnt.each do |key, value|
    ans = key if value == 1 && key > ans
  end
  ans
end
